lib/rubocop/cop/sorbet/signatures/runtime_on_failure_depends_on_checked.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Sorbet
+      # Checks that `on_failure` is not used without `checked(:tests)` or `checked(:always)`.
```

Comment:
Can you add a good/bad example as we do in other cops?

See https://github.com/Shopify/rubocop-sorbet/blob/main/lib/rubocop/cop/sorbet/forbid_t_untyped.rb for example.